src/Features/Core/Portable/ImplementInterface/AbstractImplementInterfaceService.cs

```diff
@@ -72,7 +72,54 @@ private IEnumerable<CodeAction> GetActions(Document document, ImplementTypeOptio
 
             if (state.MembersWithoutExplicitOrImplicitImplementationWhichCanBeImplicitlyImplemented.Length > 0)
             {
-                yield return ImplementInterfaceCodeAction.CreateImplementCodeAction(this, document, options, state);
+                var hasInaccessibleMembers = false;
+
+                foreach (var (_, members) in state.MembersWithoutExplicitOrImplicitImplementationWhichCanBeImplicitlyImplemented)
+                {
+                    foreach (var member in members)
+                    {
+                        switch (member)
+                        {
+                            case IPropertySymbol propertySymbol:
+
+                                if (propertySymbol.Type.DeclaredAccessibility > Accessibility.NotApplicable &&
+                                    propertySymbol.Type.DeclaredAccessibility < state.ClassOrStructType.DeclaredAccessibility)
+                                {
+                                    hasInaccessibleMembers = true;
+                                    goto AccessibilityChecked;
```

Comment:
instead of a variable + goto, just extract a local function (or sibling method) that just returns the computed value.